polygon/bridge/bridge.go

```diff
@@ -126,19 +126,30 @@ func (b *Bridge) Close() {
 // ProcessNewBlocks iterates through all blocks and constructs a map from block number to sync events
 func (b *Bridge) ProcessNewBlocks(ctx context.Context, blocks []*types.Block) error {
 	eventMap := make(map[uint64]uint64)
+	var prevSprintTime time.Time
+
 	for _, block := range blocks {
 		// check if block is start of span
 		if !b.isSprintStart(block.NumberU64()) {
 			continue
 		}
 
-		blockTimestamp := time.Unix(int64(block.Time()), 0)
-		lastDBID, err := b.store.GetSprintLastEventID(ctx, b.lastProcessedEventID, blockTimestamp, b.stateReceiverABI)
+		var timeLimit time.Time
+		if b.borConfig.IsIndore(block.NumberU64()) {
+			stateSyncDelay := b.borConfig.CalculateStateSyncDelay(block.NumberU64())
+			timeLimit = time.Unix(int64(block.Time()-stateSyncDelay), 0)
+		} else {
+			timeLimit = prevSprintTime
```

Comment:
Isn't prevSprintTime uninitialized at this point?

Reply:
It is initialized to its zero value which is 0 (and fine in this case)

Reply:
Yeah, so is a timeLimit of 0 correct for this case?

Reply:
Yes, it produces the correct maps - there are no state sync events for the first sprint.

https://github.com/erigontech/erigon/blob/7566807113d7646878c6cd302e3b18ef2b32ccfe/polygon/bor/bor.go#L1516